CHECK_POWER_COMMANDS/System_State_Class.py

```diff
@@ -0,0 +1,66 @@
+################################################################################
+#
+#  System_State_Class - class defined to manage the state of the system.
+#                       This will include whether or not the present point
+#                       in the backstop file is within a perigee passage;
+#                       whether or not a science run is underway - or 
+#                       complete, and whether or not we have somce upon a
+#                       power command was once the science run is complete.
+#
+################################################################################
+
+class System_State_Object:
+    """
+    Class defined to manage the state of the system.
+    This will include whether or not the present point
+    in the backstop file is within a perigee passage;
+    whether or not a science run is underway - or 
+    complete, and whether or not we have come upon a
+    power command  once the science run is complete.
+
+    The only method is print_system_state whcih is used for debugging.
+    """
+    def __init__(self, ):
+
+        # System state dictionary
+        self.state = {'date_cmd':  '',
+                      'time_cmd': 0,
+                      # Science Run status
+                      'science_run_exec': 'unk',
+                      # Start dates
+                      'science_run_start_date': 'unk',
+                      'science_run_start_time': 'unk',
+                      # Stop Times
+                      'science_run_stop_date': 'unk',
+                      'science_run_stop_time': 'unk',
+                      # In a perigee passage? if so what part?
+                      'perigee_passage': 'unk',
+                      # Have I run a WSPOW command after the first stop science?
+                      'post_sci_run_power_down': False,
+                      'post_sci_run_power_down_date': 'unk',
+                      'post_sci_run_power_down_time': 0,
+                      # The WSPOW0002A that should come if WSPOW00000 was
+                      # issued an hour ago and no science run is active
+                      'three_FEPs_up': False,
+                      'three_FEPs_up_date': 'unk',
+                      'three_FEPs_up_time': 0
+                      
+}
+
+ 
+    #---------------------------------------------------------------------------
+    #
+    # Method: Print System State: Print out the present values of the 
+    #                             system state parameters
+    #
+    #---------------------------------------------------------------------------
+    def print_system_state(self,):
```

Comment:
This won't run in Python 3 when the ska Python switches to that, so maybe we should do the `from __future__ import print_function` at the top and then make these calls to `print()`.